robot-server/robot_server/settings.py

```diff
@@ -103,5 +103,13 @@ class RobotServerSettings(BaseSettings):
         ),
     )
 
+    maximum_data_files: int = Field(
+        default=50,
+        gt=0,
+        description=(
+            "The maximum number of data files to allow before auto-deleting old ones."
```

Comment:
Should this say:

```suggestion
            "The maximum number of unused data files to allow before auto-deleting old ones."
```

And be named `maximum_unused_data_files`?

Reply:
Oh, and would you also mind running `pipenv run python scripts/settings_schema.py settings_schema.json` and then a top-level `make format-js`?

I'm not sure if `settings_schema.json` is really helpful, but if it exists, we should keep it up to date. I'd also be down to just delete it.

Reply:
It's actually not max number of unused data files though. As in, it will start deleting unused files when [used+unused] exceed 50.

Reply:
Does that imply that if you have 10 runs that each use 5 CSV files that are all distinct from each other, it's impossible to upload any additional CSV files?

Reply:
No, in that case we'll exceed the 50 file limit and allow adding new ones. Each time we try to add a new one, we will re-check to find any newly un-referenced files and delete them until files < 50 again.